lib/Doctrine/ORM/Mapping/ToOneOwningSideMapping.php

```diff
@@ -0,0 +1,177 @@
+
+        foreach ($mapping->joinColumns as $joinColumn) {
+            if ($mapping->isOneToOne() && ! $isInheritanceTypeSingleTable) {
+                if (count($mapping->joinColumns) === 1) {
+                    if (empty($mapping->id)) {
+                        $joinColumn->unique = true;
+                    }
+                } else {
+                    $uniqueConstraintColumns[] = $joinColumn->name;
+                }
+            }
+
+            if (empty($joinColumn->name)) {
+                $joinColumn->name = $namingStrategy->joinColumnName($mapping->fieldName, $name);
+            }
+
+            if (empty($joinColumn->referencedColumnName)) {
+                $joinColumn->referencedColumnName = $namingStrategy->referenceColumnName();
+            }
+
+            if ($joinColumn->name[0] === '`') {
+                $joinColumn->name   = trim($joinColumn->name, '`');
+                $joinColumn->quoted = true;
+            }
+
+            if ($joinColumn->referencedColumnName[0] === '`') {
+                $joinColumn->referencedColumnName = trim($joinColumn->referencedColumnName, '`');
+                $joinColumn->quoted               = true;
+            }
+
+            $mapping->sourceToTargetKeyColumns[$joinColumn->name] = $joinColumn->referencedColumnName;
+            $mapping->joinColumnFieldNames[$joinColumn->name]     = $joinColumn->fieldName ?? $joinColumn->name;
+        }
+
+        if ($uniqueConstraintColumns) {
+            if (! $table) {
+                throw new RuntimeException('ClassMetadata::setTable() has to be called before defining a one to one relationship.');
+            }
+
+            $table['uniqueConstraints'][$mapping->fieldName . '_uniq'] = ['columns' => $uniqueConstraintColumns];
+        }
+
+        $mapping->targetToSourceKeyColumns = array_flip($mapping->sourceToTargetKeyColumns);
+
+        return $mapping;
+    }
+
+    public function offsetSet(mixed $offset, mixed $value): void
+    {
+        if ($offset === 'joinColumns') {
+            $joinColumns = [];
+            foreach ($value as $column) {
+                $joinColumns[] = JoinColumnMapping::fromMappingArray($column);
+            }
+
+            $this->joinColumns = $joinColumns;
+
+            return;
+        }
+
+        parent::offsetSet($offset, $value);
+    }
+
+    /** @return array<string, mixed> */
+    public function toArray(): array
+    {
+        $array = parent::toArray();
+
+        $joinColumns = [];
+        foreach ($array['joinColumns'] as $column) {
+            $joinColumns[] = (array) $column;
+        }
+
+        $array['joinColumns'] = $joinColumns;
+
+        return $array;
+    }
+
+    /** @return list<string> */
+    public function __sleep(): array
+    {
+        $serialized = parent::__sleep();
+
+        $serialized[] = 'joinColumns';
+        $serialized[] = 'sourceToTargetKeyColumns';
+        $serialized[] = 'targetToSourceKeyColumns';
+
+        return $serialized;
```

Comment:
Not a blocker, but you could make use of the spread operator here.

```suggestion
        return [
            ...parent::__sleep(),

            'joinColumns',
            'sourceToTargetKeyColumns',
            'targetToSourceKeyColumns',
        ];
```